Horizon/Horizon/Sources/Features/Notebook/Note/Domain/UpdateNotebookNoteInteractor.swift

```diff
@@ -0,0 +1,59 @@
+//
+// This file is part of Canvas.
+// Copyright (C) 2024-present  Instructure, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+//
+
+import Combine
+
+struct CourseNoteUpdate {
+    let content: String
+    let id: String
+    let labels: [CourseNoteLabel]
+}
+
+final class UpdateNotebookNoteInteractor {
+    // MARK: Dependencies
+
+    private let courseNotesRepository: CourseNotesRepositoryProtocol
+    private var subscriptions = Set<AnyCancellable>()
+
+    // MARK: Init
+    init(courseNotesRepository: CourseNotesRepositoryProtocol) {
+        self.courseNotesRepository = courseNotesRepository
+    }
+
+    // MARK: Public
+
+    func update(
+        noteId: String,
+        content: String? = nil,
+        labels: [CourseNoteLabel]? = nil
+    ) -> Future<Void, Error> {
+        Future { [weak self] promise in
```

Comment:
Here your return `<Void, Error> ` and the function body you did't return any failure, please change it to `<Void, Never> `, 
In addition why you return `Future` please change it to `AnyPublisher<Void, Never>`

Reply:
It's a future because it doesn't need to be a publisher. It will only ever return once. On an update or delete call, we won't ever be returning multiple values through the publisher.

The Error is deliberate too. In the future when we're updating to something real, it may return an error, even though currently the hardcoded repository won't ever do that.